tests/components/modbus/test_climate.py

```diff
@@ -346,6 +347,93 @@ async def test_config_swing_mode_register(hass: HomeAssistant, mock_modbus) -> N
                 }
             ],
         },
+        {
+            CONF_CLIMATES: [
+                {
+                    CONF_NAME: TEST_ENTITY_NAME,
+                    CONF_TARGET_TEMP: 23,
+                    CONF_ADDRESS: 23,
+                    CONF_SLAVE: 1,
+                    CONF_MIN_TEMP: 16,
+                    CONF_MAX_TEMP: 23,
+                    CONF_HVAC_ONOFF_REGISTER: 0,
+                    CONF_HVAC_ONOFF_REGISTER_WRITE_TYPE: "write_coil",
+                }
+            ],
+        },
+        {
+            CONF_CLIMATES: [
+                {
+                    CONF_NAME: TEST_ENTITY_NAME,
+                    CONF_TARGET_TEMP: 23,
+                    CONF_ADDRESS: 23,
+                    CONF_SLAVE: 1,
+                    CONF_MIN_TEMP: 16,
+                    CONF_MAX_TEMP: 23,
+                    CONF_HVAC_ONOFF_REGISTER: 0,
+                    CONF_HVAC_ONOFF_REGISTER_WRITE_TYPE: "write_register",
+                }
+            ],
+        },
+        {
+            CONF_CLIMATES: [
+                {
+                    CONF_NAME: TEST_ENTITY_NAME,
+                    CONF_TARGET_TEMP: 23,
+                    CONF_ADDRESS: 23,
+                    CONF_SLAVE: 1,
+                    CONF_MIN_TEMP: 16,
+                    CONF_MAX_TEMP: 23,
+                    CONF_HVAC_ONOFF_REGISTER: 0,
+                    CONF_HVAC_ONOFF_REGISTER_WRITE_TYPE: "write_registers",
+                }
+            ],
+        },
+        {
+            CONF_CLIMATES: [
+                {
+                    CONF_NAME: TEST_ENTITY_NAME,
+                    CONF_TARGET_TEMP: 23,
+                    CONF_ADDRESS: 23,
+                    CONF_SLAVE: 1,
+                    CONF_MIN_TEMP: 16,
+                    CONF_MAX_TEMP: 23,
+                    CONF_HVAC_ONOFF_REGISTER: 0,
+                    CONF_HVAC_ONOFF_REGISTER_WRITE_TYPE: "write_coil",
+                    CONF_WRITE_REGISTERS: True,
+                }
+            ],
+        },
+        {
+            CONF_CLIMATES: [
+                {
+                    CONF_NAME: TEST_ENTITY_NAME,
+                    CONF_TARGET_TEMP: 23,
+                    CONF_ADDRESS: 23,
+                    CONF_SLAVE: 1,
+                    CONF_MIN_TEMP: 16,
+                    CONF_MAX_TEMP: 23,
+                    CONF_HVAC_ONOFF_REGISTER: 0,
+                    CONF_HVAC_ONOFF_REGISTER_WRITE_TYPE: "write_register",
+                    CONF_WRITE_REGISTERS: True,
+                }
+            ],
+        },
+        {
+            CONF_CLIMATES: [
+                {
+                    CONF_NAME: TEST_ENTITY_NAME,
```

Comment:
I do not think this is sufficient testing, you need to secure that all code is passed since we have coverage 100%.

Reply:
I got some bot comments on uncovered code indeed, will work on that,

Reply:
Added some more test cases, it is now back at 99% for the climate entity

It's not at 100 because the "None" result is/was not covered
https://github.com/home-assistant/core/blob/dev/homeassistant/components/modbus/climate.py#L492
fixing that is out of this scope

N.B. the only way to proof that is to mark it ready for review again, but I'll sleep a night over that so I can think about a way to address your concerns